library/std/arithmetic.qs

```diff
@@ -488,4 +488,74 @@ namespace Microsoft.Quantum.Arithmetic {
         }
     }
 
+    /// # Summary
+    /// Sets a zero-initialized little-endian register zs to the sum of
+    /// little-endian registers xs and ys using the carry-lookahead algorithm.
+    ///
+    /// # Description
+    /// Computes zs := xs + ys + zs[0] modulo 2ⁿ, where xs, ys, and zs are
+    /// little-endian registers, Length(xs) = Length(ys) ≤ Length(zs) = n (!!!),
+    /// assuming zs is 0-initialized, except for maybe zs[0], which can be
+    /// in |0> or |1> state and can be used as carry-in.
+    /// NOTE: `zs[Length(xs)]` can be used as carry-out, if `zs` is longer than `xs`.
+    /// This operation uses the carry-lookahead algorithm.
+    ///
+    /// # Reference
+    ///     - [arXiv:quant-ph/0406142](https://arxiv.org/abs/quant-ph/0406142)
+    ///      "A logarithmic-depth quantum carry-lookahead adder" by
+    ///      Thomas G. Draper, Samuel A. Kutin, Eric M. Rains, Krysta M. Svore
+    operation LookAheadAddLE(xs : Qubit[], ys : Qubit[], zs : Qubit[]) : Unit is Adj {
+        let xsLen = Length(xs);
+        let zsLen = Length(zs);
+        Fact(Length(ys) == xsLen, "Registers `xs` and `ys` must be of same length.");
+        Fact(zsLen == xsLen or zsLen == xsLen + 1, "Register `zs` must be same length as `xs` or one bit longer.");
+
+        if zsLen == xsLen + 1 { // with carry-out
+            // compute initial generate values
+            for k in 0..xsLen - 1 {
+                ApplyAndAssuming0Target(xs[k], ys[k], zs[k + 1]);
```

Comment:
A quantum AND gate always assumes that the target qubit is |0>, therefore the name is somewhat redundant. I suggest to use `ApplyAnd` or `CCNOTAssuming0Target`.

Reply:
We will fix this and similar places when we have proper intrinsic for this function. We'll probably move it to Canon and just call it AND gate.